[Lim Xin Yi Nicole] iP

nicolelim02 opened the pull request:
## DukePro

> “Your mind is for having ideas, not holding them.” – David Allen

DukePro frees your mind of having to remember things you need to do. It's,

- Text-based
- Easy to learn
- ~FAST~_SUPER FAST_ to use

All you need to do is,

1. Download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)
2. Double click it
3. Add your tasks
4. Let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

nicolelim02 committed:
chatbot greets user, echoes after user and exits when users type "bye"

nicolelim02 committed:
users can add task to the task list

nicolelim02 committed:
users can see all the tasks added to the tasks list

nicolelim02 committed:
users can mark a task as done or not done

nicolelim02 committed:
users can add todos

nicolelim02 committed:
users can add deadlines

nicolelim02 committed:
users can add events

nicolelim02 committed:
exception is thrown if description is empty, date isn't specified or invalid commands are provided

nicolelim02 committed:
users can delete tasks

nicolelim02 committed:
throws an exception if user does not specify the task number or task number exceed the number of task

nicolelim02 committed:
changed the ui

nicolelim02 committed:
added ui test cases

nicolelim02 committed:
refactored code

nicolelim02 committed:
Refactor code

nicolelim02 committed:
Handle commands using enums

nicolelim02 committed:
Handle commands using enums

nicolelim02 committed:
Remove irrelevant imports

nicolelim02 committed:
Remove irrelevant catch block

nicolelim02 committed:
Save tasks / changes into a text file

nicolelim02 committed:
Store dates as java.time.LocalDate

nicolelim02 committed:
Get tasks for a specific date

nicolelim02 committed:
Alert users to erroneous date entry

nicolelim02 committed:
Merge branch 'branch-Level-8' and resolve merge conflict

nicolelim02 committed:
Refactor code to extract out Ui, Storage, Parser and TaskList classes

nicolelim02 committed:
Organise classes into suitable packages

nicolelim02 committed:
Fix bugs in mark, unmark and delete commands

nicolelim02 committed:
Add existing packages into a package duke

nicolelim02 committed:
Modify the shell script to run test cases

nicolelim02 committed:
Merge commit '556af3f47a96b32898ab4cdbd65b16486a4871e8'

nicolelim02 committed:
Add JUnit tests to test the behaviour of the code

nicolelim02 committed:
Modify build.gradle

nicolelim02 committed:
Add javadocs

nicolelim02 committed:
Add javadocs for classes

nicolelim02 committed:
Add checkstyle

nicolelim02 committed:
Reorder import statements

nicolelim02 committed:
Find tasks based on the keyword provided

nicolelim02 committed:
Merge branch 'branch-A-CodingStandard'

nicolelim02 committed:
Merge branch 'branch-Level-9'

nicolelim02 committed:
Tweak code to comply with the coding standard

nicolelim02 committed:
Add a GUI to Duke

nicolelim02 committed:
Modify user interface

nicolelim02 committed:
Modify code to comply with the coding standard

nicolelim02 committed:
Add varargs to find command

nicolelim02 committed:
Add varargs to list command

nicolelim02 committed:
Merge branch 'branch-Level-10'

carriezhengjr commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,81 @@
+    }
+
+    /**
+     * Echoes user's input
+     * @param message A message to echo
+     * @return The message user's provided
```
You could leave an empty line between description and parameter section as suggested here: https://se-education.org/guides/conventions/java/intermediate.html#comments 
I will just highlight one here 😃

carriezhengjr commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,81 @@
+        }
+    }
+
+    public String getResponse(String input) {
```
Perhaps you could add JavaDoc for this method with @param and @return 🤔

carriezhengjr commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,28 @@
+     */
+    public abstract boolean isExit();
+
+    /**
+     * Executes the Command
+     * @param ui Prints the messages based on the type of Command
+     * @param storage Saves the modified list of tasks
+     * @param taskList List of tasks
+     * @throws DukeException if invalid inputs are provided
+     */
```
Clear documentation especially for param descriptions! 👍

carriezhengjr commented on src/main/java/duke/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,51 @@
+     * @return True if it is an Exit Command
+     */
+    @Override
+    public boolean isExit() {
```
Good name for a method that returns a boolean 👍

carriezhengjr commented on src/main/java/duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,46 @@
+
+    /**
+     * Constructs a new Delete Command
+     * @param taskNumber The task to delete
```
Could be "The index of the task to delete"? 🤔

carriezhengjr commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,30 @@
+        return false;
+    }
+
+    @Override
+    public String execute(Ui ui, Storage storage, TaskList taskList) {
+        ArrayList<Task> list = taskList.find(this.keyword);
+        return ui.printFind(list);
+    }
```
Perhaps you could add a JavaDoc comment to this methods? 🤔

carriezhengjr commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,99 @@
+public class Parser {
+
+    /**
+     * Checks if the input is valid
```
Perhaps this method is more like making sense of user input, while the method below checks for validity of input. 🤔

carriezhengjr commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,140 @@
+import duke.tasks.ToDo;
+
+/**
+ * This class encapsulates the Storage
```
Perhaps you could give more description on what the class does, e.g. in this case "... which stores tasks into the local hard disk". 🤔 

Same for the other classes. 😄

carriezhengjr commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,187 @@
+        this.scanner = new Scanner(System.in);
+    }
+
+    /**
+     * Greets the user whenever the application starts
+     */
```
Missing @return ? 🤔

carriezhengjr commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,187 @@
+        return "Hello! I'm Duke\nWhat can I do for you? ^_^";
+    }
+
+    /**
+     * Exits when the user chooses to stop the application
+     */
```
Missing @return ? 
Same for other methods below. 🤔

carriezhengjr commented on src/test/java/duke/ParserTest.java:
```diff
@@ -0,0 +1,37 @@
+import duke.data.exception.DukeException;
+import duke.parser.Parser;
+
+public class ParserTest {
```
Could include JavaDoc comments for this class? Perhaps same for other test classes? 🤔 
(https://se-education.org/guides/conventions/java/intermediate.html#comments)

carriezhengjr reviewed (commented):
Looks good, overall well-documented and comprehensive! 👍 
Just some nits on coding standard, mainly JavaDoc. 😄

nicolelim02 committed:
Merge branch 'branch-A-Varargs'

nicolelim02 committed:
Remove tasks command

nicolelim02 committed:
Refactor code

nicolelim02 committed:
Refactor code

nicolelim02 committed:
Add streams to filter tasks

Using streams help to reduce the level of
nesting in the code.

Filtering a stream allows for a cleaner and more readable code.

nicolelim02 committed:
Modify code to comply with coding standards

Santosh3007 commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,48 @@
+ * This class encapsulates an event item
+ */
+public class Event extends Task {
+    protected LocalDate at;
```
Maybe this variable could be named `timing` or `time` because `at` is a bit ambiguous and might mean location as well?

Santosh3007 commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,31 @@
+ * the keywords provided
+ */
+public class FindCommand extends Command {
+    private String[] keyword;
```
Since this is an array of strings maybe it could be named `keywords` making it plural?

Santosh3007 commented on src/main/java/duke/commands/ListCommand.java:
```diff
@@ -0,0 +1,58 @@
+     * @param info Essential information for the output list
+     */
+    public ListCommand(String... info) {
+        super();
```
A small thing but I don't think you have to call the superclass's constructor explicitly if it does not take in any arguments because by default it will always call `super()` if there is no call to the superclass's constructor :)

Santosh3007 reviewed (commented):
Looks good and not much code quality violations I could find! :)

nicolelim02 committed:
Add assert statements

nicolelim02 committed:
Add assert statements

nicolelim02 committed:
Refactor code

nicolelim02 committed:
Refactor code

nicolelim02 committed:
Enable assertions

The Storage and Parser class deals with external
inputs provided by the user.

Enabling assertions help to check if the inputs
provided by the user is valid.

Assertions provide a simple and clean way to
validate any assumptions made in the program.

nicolelim02 committed:
Merge pull request #1 from nicolelim02/branch-A-Assertions

Branch-A-Assertions

nicolelim02 committed:
Merge branch 'master' into branch-A-CodeQuality

nicolelim02 committed:
Merge pull request #2 from nicolelim02/branch-A-CodeQuality

nicolelim02 committed:
Merge branch 'master' into branch-A-Streams

nicolelim02 committed:
Merge pull request #3 from nicolelim02/branch-A-Streams

nicolelim02 committed:
Fix bugs in Ui

nicolelim02 committed:
Set up CI

nicolelim02 committed:
Fix checkstyle and test cases

nicolelim02 committed:
Provides users with a summary of their completed and upcoming tasks

Tracks the task's completion status and when was is it marked as completed.

This helps users to keep track of what needs to be done and what has been
completed so far.

nicolelim02 committed:
Improve GUI

nicolelim02 committed:
Rename chatbot

nicolelim02 committed:
Remove unnecessary code

nicolelim02 committed:
Modify build.gradle file

nicolelim02 committed:
Update README

Commit:
Remove unused code

Commit:
Refactor code

Commit:
Improve error handling

nicolelim02 committed:
Rename a command

nicolelim02 committed:
Update error messages

nicolelim02 committed:
Update README

nicolelim02 committed:
Add a screenshot of the product

nicolelim02 committed:
Add more JUnit tests

nicolelim02 committed:
Close chatbot after bye

nicolelim02 committed:
Fix order of import statements

nicolelim02 committed:
Refactor code

nicolelim02 committed:
Refactor code

nicolelim02 committed:
Change user guide's theme

nicolelim02 committed:
Update user guide's theme
